dace/transformation/testing.py

```diff
@@ -68,7 +68,9 @@ def _optimize_recursive(self, sdfg: SDFG, depth: int):
                 print('    ' * depth, type(match).__name__, '- ', end='', file=self.stdout)
 
                 tsdfg: SDFG = new_sdfg.sdfg_list[match.sdfg_id]
-                match.apply(tsdfg)
+                tgraph = tsdfg.node(match.state_id) if match.state_id >= 0 else tsdfg
+                match._sdfg = tsdfg
```

Comment:
Why do we need to set the (private) attribute manually here, since it is anyway passed through the `apply` method? Alternatively, if we must do this, wouldn't it be better to make the attribute public?

Reply:
That's because the optimizer deepcopies the SDFG.

Reply:
Good point though.